src/arch/linux/async/signal.c

```diff
@@ -437,10 +440,9 @@ static void leavedos_signal(int sig, siginfo_t *si, void *uc)
 {
   ucontext_t *uct = uc;
   sigcontext_t *scp = &uct->uc_mcontext;
-  init_handler(scp, uct->uc_flags);
+  init_handler0(scp);
```

Comment:
Why?
Don't we need full init_handler() for anything
that can come from native dpmi?